codespell_lib/_codespell.py

```diff
@@ -228,7 +228,6 @@ def open_with_chardet(self, filename: str) -> Tuple[List[str], str]:
                     break
         self.encdetector.close()
         encoding = self.encdetector.result["encoding"]
-        assert encoding is not None  # noqa: S101
```

Comment:
The assertion was introduced by `8d0d82b` / #2588 in an attempt to please mypy. That's not a proper fix, especially since Python can optimise assertions away, but mypy did give a good hint to look into this part of the code. And we now have the proper fix!